### PR TITLE
feat: 1520 404 page not found screen development

### LINKED ISSUE #1520: 404 page not found screen development
**Describe the task**
Developing the error screen that appears in SPAR when the page or content the user is trying to access cannot be found.

**Acceptance Criteria**
- [ ] Peer review with a developer
- [ ] QA review with designers
- [ ] Merge PR

**Additional context**
**Need to complete #1519 prior to starting this one
**Text for the smallest screen breakpoint (320 px):**

Page not found
This page doesn't exist or was removed.

**Text for every other breakpoint:**

Access denied

Oops! It looks like this page got lost in the forest.
This page doesn't exist or was removed.

---

The Figma file with screen designs for reference can be found [here](https://www.figma.com/design/XpF15ggH9Gn7FeXa9zTNYm/SPAR---404-and-403-errors?node-id=4001-19244&t=f6cpw1jAUnQRrejv-1).

This is the .svg image file:

Image

### PR DESCRIPTION
**Pull request by ngunner15**
# Description
Closes #1520 
Test URL for 404 page - https://nr-spar-13-frontend.apps.silver.devops.gov.bc.ca/404

#### New
- Route for 404
- Text for 404 page

#### Changed
- Route file

#### Removed
-

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [ ] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-13-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1663-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1663-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-spar-13-frontend.apps.silver.devops.gov.bc.ca/)
- [Backend](https://nr-spar-1663-backend.apps.silver.devops.gov.bc.ca/swagger-ui/index.html)
- [Oracle-API](https://nr-spar-1663-oracle-api.apps.silver.devops.gov.bc.ca/actuator/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-spar/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge.yml)

### CODE REVIEW

**Review by RMCampos — APPROVED** [review]
Great job Nisarg! 💯 🥇 

LGTM!

**Comment by RMCampos** [comment]
Working like a charm

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -99,6 +104,11 @@ const App: React.FC = () => {
     browserRouter.navigate('/403');
   };
 
+  // Programmatic redirection to 404
+  // const handleRedirectTo404 = () => {
+  //   browserRouter.navigate('/404');
+  // };
```
Comment (mgaseta) [inline frontend/src/App.tsx:110]: Remove this comment.

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -90,6 +90,11 @@ const App: React.FC = () => {
   const getBrowserRouter = () => {
     const selectedRoutes = selectBrowserRoutes();
     selectedRoutes.push(...sharedRoutes);
+    // Add a wildcard route to catch undefined routes and show the 404 page
+    selectedRoutes.push({
+      path: '*',
+      element: <FourOhFour />
+    });
```
Comment (mgaseta) [inline frontend/src/App.tsx:97]: Why is this necessary? Shouldn't the `signedRoutes` const be enough to handle the 404 path? Also, you can use the ROUTES constant to get the `'*'` path.

**Review by mgaseta — CHANGES_REQUESTED** [review]
Looking good! Nice job!
Besides the markups, there are 2 issues:
1. On the design, the navigation header is still present on the 404 page ([here](https://www.figma.com/design/XpF15ggH9Gn7FeXa9zTNYm/SPAR---404-and-403-errors?node-id=4001-19343&node-type=frame&t=VyNolVOCxnnnMnsR-0))
2. The page is not being redirected to the 404, you can see that by looking at the URL, it doesn't change to `/404`, not exactly necessary, but it would be nice to change the URL to reflect the correct page.

File: frontend/src/App.tsx (hunk 2 above)
Comment (ngunner15) [reply frontend/src/App.tsx]: https://github.com/bcgov/nr-spar/pull/1663/commits/58b4202bcaa9010b696f0fdda1b12ecd2d68d268

**Comment by kevinginley** [comment]
Looks good! Responsiveness working well. Thanks Nisarg

**Review by mgaseta — APPROVED** [review]
LGTM! Nice job @ngunner15!

**Pull request merged**